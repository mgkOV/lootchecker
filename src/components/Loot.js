import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchBitcoin } from '../actions/bitcoin';

export class Loot extends Component {
  static propTypes = {
    balance: PropTypes.number.isRequired,
    bitcoin: PropTypes.object.isRequired,
    fetchBitcoin: PropTypes.func.isRequired
  };

  componentDidMount() {
    this.props.fetchBitcoin();
  }

  computeBitcoin = () => {
    const { bitcoin, balance } = this.props;

    if (Object.keys(bitcoin).length === 0) return '';
    const usdRate = parseInt(bitcoin.bpi.USD.rate.replace(',', ''), 10);
    return balance / usdRate;
  };

  render() {
    return <h3>Bitcoin balance: {this.computeBitcoin()}</h3>;
  }
}

export default connect(({ balance, bitcoin }) => ({ balance, bitcoin }), {
  fetchBitcoin
})(Loot);
