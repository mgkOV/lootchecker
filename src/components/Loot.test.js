import React from 'react';
import { mount, shallow } from 'enzyme';
import { Loot } from './Loot';

describe('Loot', () => {
  const mockFetchBitcoin = jest.fn();
  const props = {
    balance: 10,
    bitcoin: { bpi: { USD: { rate: '1,000' } } },
    fetchBitcoin: mockFetchBitcoin
  };
  const loot = shallow(<Loot {...props} />);

  it('renders properly', () => {
    expect(loot).toMatchSnapshot();
  });

  describe('when mounted', () => {
    const loot = mount(<Loot {...props} />);

    it('dispatches the "fetchBitcoin()" method it recives from props', () => {
      expect(mockFetchBitcoin).toHaveBeenCalled();
    });
  });

  describe('when there are valid bitcoin props', () => {
    it('displays the correct bitcoin value', () => {
      expect(loot.find('h3').text()).toEqual('Bitcoin balance: 0.01');
    });
  });
});
