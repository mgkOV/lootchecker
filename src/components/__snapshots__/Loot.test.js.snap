// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loot
    balance={10}
    bitcoin={
        Object {
            "bpi": Object {
              "USD": Object {
                "rate": "1,000",
              },
            },
          }
    }
    fetchBitcoin={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Bitcoin balance: ",
        0.01,
      ],
    },
    "ref": null,
    "rendered": Array [
      "Bitcoin balance: ",
      0.01,
    ],
    "type": "h3",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Bitcoin balance: ",
          0.01,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Bitcoin balance: ",
        0.01,
      ],
      "type": "h3",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
